bridgesync/backfill_tx_sender.go

```diff
@@ -0,0 +1,272 @@
+		SELECT block_num, block_pos, tx_hash
+		FROM %s
+		WHERE txn_sender = '' OR txn_sender IS NULL
+		ORDER BY block_num, block_pos
+		LIMIT $1 OFFSET $2
+	`, tableName)
+
+	rows, err := b.db.QueryContext(ctx, query, limit, offset)
+	if err != nil {
+		return nil, fmt.Errorf("failed to query records needing backfill: %w", err)
+	}
+	defer rows.Close()
+
+	var records []RecordToBackfill
+	for rows.Next() {
+		var record RecordToBackfill
+		var txHashStr string
+
+		err := rows.Scan(&record.BlockNum, &record.BlockPos, &txHashStr)
+		if err != nil {
+			return nil, fmt.Errorf("failed to scan record: %w", err)
+		}
+
+		record.TxHash = common.HexToHash(txHashStr)
+		records = append(records, record)
+	}
+
+	return records, nil
+}
+
+// processBatch processes a batch of records to backfill txn_sender
+func (b *BackfillTxnSender) processBatch(
+	ctx context.Context,
+	tableName string,
+	records []RecordToBackfill,
+) {
+	// First, extract all txn_sender data
+	updates := make([]RecordUpdate, 0, len(records))
+	var failedExtractions []string
+
+	for _, record := range records {
+		// Extract txn_sender from transaction hash
+		txnSender, err := b.extractTxnSender(record.TxHash)
+		if err != nil {
+			b.log.Errorf("Failed to extract txn_sender for tx %s: %v", record.TxHash.Hex(), err)
+			failedExtractions = append(failedExtractions, record.TxHash.Hex())
+			continue
+		}
+
+		updates = append(updates, RecordUpdate{
+			BlockNum:  record.BlockNum,
+			BlockPos:  record.BlockPos,
+			TxnSender: txnSender,
+		})
+	}
+
+	// Perform bulk update if we have any successful extractions
+	if len(updates) > 0 {
+		if err := b.bulkUpdateTxnSender(ctx, tableName, updates); err != nil {
+			b.log.Errorf("Failed to bulk update txn_sender for %d records: %v", len(updates), err)
+		} else {
+			b.log.Infof("Successfully bulk updated txn_sender for %d records", len(updates))
+		}
+	}
+
+	// Log failed extractions
+	if len(failedExtractions) > 0 {
+		b.log.Errorf("Failed to extract txn_sender for %d transactions: %v", len(failedExtractions), failedExtractions)
+	}
+}
+
+// extractTxnSender extracts the transaction sender from a transaction hash
+func (b *BackfillTxnSender) extractTxnSender(txHash common.Hash) (common.Address, error) {
+	// Use the new extractCallData function to get the transaction sender
+	_, rootCall, err := extractCallData(b.client, b.bridgeAddr, txHash, true, b.log)
+	if err != nil {
+		return common.Address{}, fmt.Errorf("failed to extract root call: %w", err)
+	}
+
+	return rootCall.From, nil
+}
+
+// bulkUpdateTxnSender performs a bulk update of txn_sender for multiple records
+func (b *BackfillTxnSender) bulkUpdateTxnSender(
+	ctx context.Context,
+	tableName string,
+	updates []RecordUpdate,
+) error {
+	if len(updates) == 0 {
+		return nil
+	}
+
+	tx, err := b.db.BeginTx(ctx, nil)
+	if err != nil {
+		return fmt.Errorf("failed to bulk update txn_sender: %w", err)
+	}
+	defer func() {
+		if err != nil {
+			_ = tx.Rollback()
```

Comment:
Can we at least log the `err` and the rollback error?

Use [this](https://github.com/agglayer/aggkit/blob/8b7c1b1cd10f041b1675e101aca1b3daab74a841/l2gersync/processor.go#L95) as an example. WDYT?